app/src/lib/git/checkout.ts

```diff
@@ -105,7 +100,17 @@ export async function checkoutBranch(
     progressCallback
   )
 
-  await git(args, repository.path, 'checkoutBranch', opts)
+  await withTrampolineEnvForRemoteOperation(
+    account,
+    getFallbackUrlForProxyResolve(account, repository),
+    env => {
+      return git(args, repository.path, 'checkoutBranch', {
+        ...opts,
+        env: merge(opts.env, env),
```

Comment:
Commands that also have a `progressCallback` need this additional step of merging the `env` provided by `withTrampolineEnvForRemoteOperation` and the one coming from that progress callback.